src/app/clusters/joint-fabric-administrator-server/joint-fabric-administrator-server.cpp

```diff
@@ -158,19 +205,70 @@ bool emberAfJointFabricAdministratorClusterTransferAnchorCompleteCallback(
     return true;
 }
 
-// TODO
 bool emberAfJointFabricAdministratorClusterAnnounceJointFabricAdministratorCallback(
     chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
     const chip::app::Clusters::JointFabricAdministrator::Commands::AnnounceJointFabricAdministrator::DecodableType & commandData)
 {
     MATTER_TRACE_SCOPE("AnnounceJointFabricAdministrator", "JointFabricAdministrator");
 
+    ChipLogProgress(JointFabric, "emberAfJointFabricAdministratorClusterAnnounceJointFabricAdministratorCallback: %u",
+                    commandData.endpointID);
+
+    auto nonDefaultStatus = Status::Success;
+
+    auto & failSafeContext = Server::GetInstance().GetFailSafeContext();
+    VerifyOrExit(failSafeContext.IsFailSafeArmed(commandObj->GetAccessingFabricIndex()),
```

Comment:
Again: this is not in the spec.  Why is this check here?  I am comparing this code to "11.25.7.9. AnnounceJointFabricAdministrator Command".